snapshot-labs/sx-monorepo
# fix: fix error when auto logging with locked metamask

### Summary



Closes: #185 

Fix issue when an auto-log user ignored the metamask unlock request, and refresh the page.
The case where the whole page crash described in the issue has been resolved via another PR.

### How to reproduce (before this PR)

1. Go to https://snapshotx.xyz/#/s:ens.eth
2. Log in with your metamask wallet
3. Go to metamask, and lock it
4. Refresh the page
5. The ACCOUNT button in the TopNav (upper-right corner) should be in a loading state, and metamask should prompt you to unlock it with your password.
6. Ignore the metamask prompt
7. All actions on the website should still be considered as guest

Everything is working as intended until now.

1. Now, refresh the page
2. Continue to ignore the metamask prompt
3. Open the Dev console
4. You should see some errors about stamp not being able to fetch the name for the logged account 

This is due to this promise

https://github.com/snapshot-labs/sx-monorepo/blob/407d18434fc32a5c623690512d96acf84f4e0507/apps/ui/src/composables/useWeb3.ts#L32

not being awaited anymore, and metamask just erroring with `message: "Already processing eth_requestAccounts. Please wait."`, continuing the flow, and switching `state.authLoading` to `false`, and make the TopNav assuming that the user is now logged in.

### How to test (with this PR)

1. Follow the same steps described above
2. When refreshing the page with metamask locked, it should not crash with stamp error, as the `state.authLoading` still is on `true`.
3. Unlock metamask
4. You should now be logged in
